### PR TITLE
Consolidate Compiler ErrorMessage tests under one suite.

### PR DESCRIPTION
**Pull request by vzarytovskii**
As part of #7075

### CODE REVIEW

File: tests/fsharpqa/Source/test.lst [hunk 1]
```diff
@@ -267,7 +267,6 @@ Misc01			Libraries\Core\Reflection
 Misc01			Libraries\Core\Unchecked
 Misc01			Warnings
 Misc01			ErrorMessages\UnitGenericAbstractType
```
Comment (KevinRansom) [inline tests/fsharpqa/Source/test.lst:269]: Probably no longer necessary

File: tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj [hunk 2]
```diff
@@ -5,6 +5,8 @@
   <PropertyGroup>
     <TargetFrameworks>net472;netcoreapp3.1</TargetFrameworks>
     <TargetFrameworks Condition="'$(OS)' == 'Unix'">netcoreapp3.1</TargetFrameworks>
+    <RuntimeIdentifiers>win-x86;win-x64;linux-x64;osx-x64</RuntimeIdentifiers>
```
Comment (KevinRansom) [inline tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj:8]: Probably if target framework has no windows, then the rids won't have win- either.  See line above ...
 <TargetFrameworks Condition="'$(OS)' == 'Unix'">netcoreapp3.1</TargetFrameworks>

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs [hunk 3]
```diff
@@ -1,41 +1,39 @@
...
+    [<Fact>]
     let ``Comma In Rec Ctor``() =
         CompilerAssert.TypeCheckWithErrors
             """
 type Person = { Name : string; Age : int; City : string }
 let x = { Name = "Isaac", Age = 21, City = "London" }
             """
             [|
-                FSharpErrorSeverity.Error, 1, (3, 18, 3, 52), "This expression was expected to have type\n    'string'    \nbut here has type\n    ''a * 'b * 'c'    \r\nA ';' is used to separate field values in records. Consider replacing ',' with ';'."
+                FSharpErrorSeverity.Error, 1, (3, 18, 3, 52), "This expression was expected to have type\n    'string'    \nbut here has type\n    ''a * 'b * 'c'    " + System.Environment.NewLine + "A ';' is used to separate field values in records. Consider replacing ',' with ';'."
```
Comment (KevinRansom) [inline tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs:32]: System.Environment.Newline perhaps?  There are many other places too.  Maybe \n is correct.

File: tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj (hunk 2 above)
Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/FSharp.Compiler.ComponentTests.fsproj:8]: I will make it conditional as well. Now, we target net47 by default and netcoreapp if nix.

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs (hunk 3 above)
Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs:32]: I’ve noticed that it is mixed all over the place. I usually test it on both windows and Linux and use the NewLine where applicable.

**Comment by smoothdeveloper** [comment]
In my experience, tests asserting error messages are easier to maintain when they use the baseline files.

Adjusting the test code when they are using `CompilerAssert.CompileWithErrors` is more tedious than running a diff on .err and .bsl files. It is also true for authoring new tests, and making sure the test covers all the compiler output, not just cherry picking on subset of messages (which can still hide potential regressions).

I've added `TypeCheckWithErrorsAndOptionsAgainstBaseLine` to make it convenient to use the baseline system under the new testing infrastructure, it may be worth for new tests asserting on error messages to use this system instead:

* tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConfusingTypeName.fs
* tests/FSharp.Compiler.ComponentTests/ErrorMessages/AccessOfTypeAbbreviationTests.fs

https://github.com/dotnet/fsharp/blob/d53716c48c1a9b78cb249cadb6065758122bd686/tests/FSharp.Test.Utilities/CompilerAssert.fs#L449-L478

This would also lend well into something where we cover all actual error messages the compiler can produce (which isn't the case, we've seen a regression on match on DU cases), which can also be useful for future work on compiler documentation (description of each error message with samples and explanation to fix).

**Comment by baronfel** [comment]
regarding the last point about compiler error message documentation I 100% took samples and phrasing from the baseline tests because they were such an easy to read resource.

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/InvalidNumericLiteralTests.fs [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
-namespace FSharp.Compiler.ComponentTests
+namespace FSharp.Compiler.ErrorMessages.ComponentTests
 
 open Xunit
 open FSharp.Test.Utilities
 open FSharp.Compiler.SourceCodeServices
 
-module ``Invalid Numeric Literal`` =
+module ``Numeric Literlas`` =
```
Comment (ThorstenReichert) [inline tests/FSharp.Compiler.ComponentTests/ErrorMessages/InvalidNumericLiteralTests.fs:9]:
```suggestion
module ``Numeric Literals`` =
```

**Comment by vzarytovskii** [comment]
@smoothdeveloper 

> In my experience, tests asserting error messages are easier to maintain when they use the baseline files.
> 
> Adjusting the test code when they are using `CompilerAssert.CompileWithErrors` is more tedious than running a diff on .err and .bsl files. It is also true for authoring new tests, and making sure the test covers all the compiler output, not just cherry picking on subset of messages (which can still hide potential regressions).
> 
> I've added `TypeCheckWithErrorsAndOptionsAgainstBaseLine` to make it convenient to use the baseline system under the new testing infrastructure, it may be worth for new tests asserting on error messages to use this system instead:
> 
> * tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConfusingTypeName.fs
> * tests/FSharp.Compiler.ComponentTests/ErrorMessages/AccessOfTypeAbbreviationTests.fs

It makes sense to me, I wouldn't do it as a part of this PR thought, this one is just moving tests from older suites to a new one.
I will make it as a separate though.

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/InvalidNumericLiteralTests.fs (hunk 4 above)
Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/ErrorMessages/InvalidNumericLiteralTests.fs]: Thanks, I probably need to set up a spellchecker 🤦

**Comment by smoothdeveloper** [comment]
@vzarytovskii sorry I didn't catch the tests I mentioned were just moved. 

What I say makes most sense for new tests around error / warnings, or at the time change to the actual message makes the test red.

**Pull request merged**